php-checks/src/main/java/org/sonar/php/checks/RedundantComplementOperatorCheck.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import org.sonar.check.Rule;
+import org.sonar.php.checks.utils.CheckUtils;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.expression.UnaryExpressionTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+@Rule(key = "S2761")
+public class RedundantComplementOperatorCheck extends PHPVisitorCheck {
+
+  private static final String MESSAGE = "Use the \"{!|~}\" operator just once or not at all. ";
+
+  private static final Kind[] COMPLEMENT_BINARY = {
+    Kind.BITWISE_COMPLEMENT,
+    Kind.LOGICAL_COMPLEMENT};
```

Comment:
Shouldn't we also cover the unary `-` operator?

Reply:
`--` has a different meaning than `!!` or `~~.` I thought about that too, but it's more in a meta-mathematical context. It would at most `-(-$a)` make sense to check.

Reply:
`--` is an integer casting an decrease the int before use